state_transfer/transfer/rsync/server.go

```diff
@@ -91,6 +94,14 @@ func createRsyncServerResources(c client.Client, r *RsyncTransfer, ns string) er
 
 func createRsyncServerConfig(c client.Client, r *RsyncTransfer, ns string) error {
 	var rsyncConf bytes.Buffer
+	runRsyncRoot := false
+	for _, ops := range r.options.DestContainerMutations {
+		if ops.SecurityContext().RunAsUser != nil && *ops.SecurityContext().RunAsUser == int64(0) {
```

Comment:
We also need to check `securityContext.Privileged` field. If it's set, root should be enabled.